test/context/directory/context.test.js

```diff
@@ -15,11 +15,11 @@ describe('#directory context validation', () => {
     const context = new Context({ AUTH0_INPUT_FILE: dir });
     await context.load();
 
-    expect(context.assets.rules).to.deep.equal([]);
-    expect(context.assets.databases).to.deep.equal([]);
-    expect(context.assets.pages).to.deep.equal([]);
-    expect(context.assets.clients).to.deep.equal([]);
-    expect(context.assets.resourceServers).to.deep.equal([]);
+    Object.entries(context.assets).forEach(([ k, v ]) => {
+      if (typeof v === 'undefined') delete context.assets[k];
+    });
+
```

Comment:
Rather than checking the value of each asset collection and delete any that are undefined.    The advantage of this over the previous tests is that if there is a new resource type added one would not have to update this test to ensure the correct behavior on that resource type since we iterate over all resource assets rather than hard coding the ones we want to check.